### DISCUSSION TITLE
Nuxt 3, Extra prefetch data when generating a dynamic pages

### DISCUSSION THREAD

**Discussion by dragoneknp** [nuxt/nuxt#18444] [Questions]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.17.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@8.15.0`
- Builder:          `vite`

### Reproduction

repo with reproduction:
https://github.com/dragoneknp/nuxt3-bug-reproduction 
Just run:
`npx nuxi generate` - After that you can see that in .output/public/1/index.html there are a lot of extra prefetch scripts
`npx nuxi start`

### Short explanation:
i am trying to dynamically load my content for each page, however if we look at any page we will see that the content is loaded for all pages

### Demonstration
I dynamically create pages with the:

```
import { defineNuxtModule } from '@nuxt/kit';

export default defineNuxtModule({
  setup(options, nuxt) {
    nuxt.hook('nitro:config', (nitroConfig) => {
      nitroConfig.prerender.routes.push(...[1, 2, 3].map((article) => `/${article}/`));
    });
  },
});
```

And than in my page -> pages/[id].vue
I dynamically fetch the content with the:

```
const route = useRoute();
const id = route.params.id;
const article = await import(`../data/${id}.json`);
```
<img width="1436" alt="Screenshot 2023-01-22 at 17 21 24" src="https://user-images.githubusercontent.com/70590182/213920705-fcb62788-3fb0-4760-a3ce-3271deb345b9.png">

And than I see that I have 3 files instead of 1. This is 1.json, 2.json, 3.json all files that I dynamically fetched, but I opened the page /3, so I want to see only payload to these page, but I see to all other pages
- 1.b30d3ef4.js -> 1.json
- 2.4ec56f60.js -> 2.json
- 3.2e69df39.js -> 3.json

**Comment by danielroe** [comment]
Because of how you are importing the data, vite registers a dynamic import dependency on all of those files, so it makes sense that these are prefetched. I would suggest instead using `$fetch` to get the data.
